cdc-accelerator/setup/installer.js

```diff
@@ -0,0 +1,142 @@
+import { spawnSync } from 'child_process'
+import fs from 'fs'
+import path from 'path'
+import { CONFIGURATION_FILES, PACKAGE_JSON_FILE_NAME, PREVIEW_DIRECTORY } from './constants.js'
+import { CDC_ACCELERATOR_DIRECTORY, CONFIG_FILENAME, Operations, PREVIEW_FILE_NAME, SRC_DIRECTORY } from '../core/constants.js'
+
+export default class Installer {
+    install(newProjectPackageJson, acceleratorInstallationPath, acceleratorPackageJson) {
+        this.generatePackageJsonProperties(newProjectPackageJson, acceleratorPackageJson)
+        this.copyConfigurationFiles(acceleratorInstallationPath)
+        this.generateConfigurationFile()
+        this.generatePreviewFile(acceleratorInstallationPath)
+
+        spawnSync('npm install', { shell: false, stdio: 'inherit' })
+    }
+
+    generatePackageJsonProperties(newProjectPackageJson, acceleratorPackageJson) {
+        this.#copyAcceleratorDependencies('devDependencies', newProjectPackageJson, acceleratorPackageJson)
+        this.#generateNpmScripts(newProjectPackageJson)
+        this.#generateJestConfiguration(newProjectPackageJson)
+        this.#copyDependency('light-server', newProjectPackageJson, acceleratorPackageJson)
+        fs.writeFileSync(PACKAGE_JSON_FILE_NAME, JSON.stringify(newProjectPackageJson, null, 4))
+    }
+
+    #copyAcceleratorDependencies(dependenciesProperty, newProjectPackageJson, acceleratorPackageJson) {
+        Object.entries(acceleratorPackageJson[dependenciesProperty]).forEach((entry) => {
+            if (!this.#containsForbiddenCharacters(entry[0])) {
+                Object.assign(newProjectPackageJson[dependenciesProperty], { [entry[0]]: entry[1] })
+            }
+        })
+    }
+
+    #containsForbiddenCharacters(dependency) {
+        const forbidden = dependency.match(/[*?:;,&|+]/)
+        return forbidden ? true : false
```

Comment:
I would simplify this to: return dependency.match(/[*?:;,&|+]/) ? true : false

Reply:
agree